### DISCUSSION TITLE
Trying to apply BC on a mesh generated by gmsh

### DISCUSSION THREAD

**Discussion by AlexSacMorane** [idaholab/moose#20740] [Q&A Meshing] [answered]
Dear all,

I am a new user of Moose so I have a basic problem. Even if I have looked on the forum and on Moose documentation, my problem is staying here.
I try to study a diffusive problem on a mesh generated by gmsh but I have the following error problem.

_*** ERROR ***
Task init_mesh is not registered to build MeshGenerator derived objects_

Moreover, I have converted my .msh to a .vtu with meshio-convert and use this .vtu in my .i but I get a new error.

_=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 24257 RUNNING AT ALEX-WORKING
=   EXIT CODE: 11
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Segmentation fault (signal 11)
This typically refers to a problem with your application.
Please see the FAQ page for debugging suggestions_

I do not understand what happened here.

I stay available, thank for all the help you can give me.
You will find in the attachments my .msh, .geo, .i
[tuto_gmsh_moose.zip](https://github.com/idaholab/moose/files/8447407/tuto_gmsh_moose.zip)

**Comment by GiudGiud** [comment, marked as answer]
`Task init_mesh is not registered to build MeshGenerator derived objects` means you are missing enclosing brackets on a mesh generator. 

So you have something like
```
[Mesh]
  type = FileMeshGenerator
```
instead of 
```
[Mesh]
  [fmg]
    type = FileMeshGenerator
```

Guillaume

**Reply by AlexSacMorane** [reply]
Thank it helps. 
I have changed a bit the simulation (I am now working on 2D). I have added the enclosing brackets. But I get again an error message.

_*** ERROR ***
map_find() error: key "26" not found in file ../src/mesh/gmsh_io.C on line 605_

You will find in the attachments my .i .msh .geo
[Test_gmsh_moose.zip](https://github.com/idaholab/moose/files/8452373/Test_gmsh_moose.zip)
I stay available

**Reply by GiudGiud** [reply]
Hello

Anything special about this mesh? What kind of elements does it contain?
I tried converting it to vtu to see what it looked like and I only saw an outer rectangular edge, no inside 2D elements?

Guillaume

**Reply by AlexSacMorane** [reply]
Hello,

Thank you for your answer. I have rechecked my .msh file. You are right the mesh has disappeared. I have exported the .msh again and it seems it is working at this time. I do not understand why on the first try nodes were not exported.

Regards
Alexandre

**Reply by GiudGiud** [reply]
ok great! Glad this works now
